core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala

```diff
@@ -279,3 +286,8 @@ private[spark] case class ConfigBuilder(key: String) {
     }
   }
 }
+
+object EffectTiming extends Enumeration {
+  type EffectTiming = Value
+  val LAUNCHING_DRIVER, DEPLOYING_APPLICATION, RUNTIME, DEPLOYING_CLUSTER, TESTING, UNKNOWN = Value
```

Comment:
Would you like to add comments to describe each of them, for me, I'm a little confuse on LAUNCHING_DRIVER, DEPLOYING_APPLICATION and DEPLOYING_CLUSTER